dotnet-datetime/TimersInCSharp/Test/TimerTests.cs

```diff
@@ -0,0 +1,123 @@
+using System.Threading;
+using System.Timers;
+
+namespace Test
+{
+    [TestClass]
+    public class TimerTests
+    {
+        [TestMethod]
+        public void WhenTimerSetUpWithMethodHandler_TimedEventFired()
+        {
+            var regex = new Regex("Elapsed event at");
+            var sw = new StringWriter();
+            Console.SetOut(sw);
+
+            var timer = new System.Timers.Timer(1000);
+            timer.Elapsed += OnEventExecution;
+            timer.Start();
+
+            Thread.Sleep(3000);
+            Assert.IsTrue(regex.Matches(sw.ToString()).Count >= 1);
+        }
+
+        public void OnEventExecution(Object? source, ElapsedEventArgs e)
+        {
+            Console.WriteLine($"Elapsed event at {e.SignalTime:G}");
+        }
+
+        [TestMethod]
+        public void WhenTimerSetUp_TimedEventFired()
+        {
+            var regex = new Regex("Elapsed event at");
+            var sw = new StringWriter();
+
+            var timer = new System.Timers.Timer(1000);
+            timer.Elapsed += (s, e) =>
+            {
+                sw.WriteLine($"Elapsed event at {e.SignalTime:G}");
+            };
+
+            timer.Start();
+
+            Thread.Sleep(3000);
+            Assert.IsTrue(regex.Matches(sw.ToString()).Count > 1);
+        }
+
+        [TestMethod]
+        public void WhenTimerSetUpWithAutoResetFalse_TimedEventFired()
+        {
+            var regex = new Regex("Elapsed event at");
+            var sw = new StringWriter();
+
+            var timer = new System.Timers.Timer(1000);
+            timer.AutoReset = false;
+            timer.Elapsed += (s, e) =>
+            {
+                sw.WriteLine($"Elapsed event at {e.SignalTime:G}");
+            };
+
+            timer.Start();
+
+            Thread.Sleep(3000);
+            Assert.IsTrue(regex.Matches(sw.ToString()).Count == 1);
+        }
+
+        [TestMethod]
+        public void WhenTimerSetUpWithAsyncHandler_TimedEventFired()
+        {
+            var regex = new Regex("Elapsed event at");
+            var sw = new StringWriter();
+
+            var timer = new System.Timers.Timer(1000);
+            timer.Elapsed += async (s, e) =>
+            {
+                await sw.WriteLineAsync($"Elapsed event at {e.SignalTime:G}");
+            };
+
+            timer.Start();
+
+            Thread.Sleep(3000);
+            Assert.IsTrue(regex.Matches(sw.ToString()).Count >= 1);
+        }
+
+        [TestMethod]
+        public void WhenTimerHandlerCausesException_ExceptionIsNotThrown()
+        {
+            var regex = new Regex("Elapsed event at");
+            var sw = new StringWriter();
+
+            var timer = new System.Timers.Timer(1000);
+            timer.Elapsed += (s, e) =>
+            {
+                sw.WriteLine($"Elapsed event at {e.SignalTime:G}");
+                throw new Exception();
+            };
+
+            timer.Start();
+            Thread.Sleep(3000);
+        }
```

Comment:
The test named _WhenTimerHandlerCausesException_ExceptionIsNotThrown_ tests that any exception thrown inside the Elapsed event handler of the Timer class is suppressed and doesn't bubble up.

The fact that the test passes proves that the exception is suppressed, otherwise the test would go red. 

In my opinion, this test makes sense as it is and doesn't need an assert but I might be wrong.

Reply:
Sure, makes sense.